### DISCUSSION TITLE
the sanboot command could not download iso files from openwrt in uefi mode

### DISCUSSION THREAD

**Discussion by chainofhonor** [ipxe/ipxe#754] [Q&A]
The following are the relevant configurations
####################################
item WEPE2.1-UEFI         Boot from WEPE2.1.ISO----UEFI-OPENWRT-11.3  
item WEPE2.1-BIOS         Boot from WEPE2.1.ISO----BIOS-OPENWRT-11.3 
item WEPE2.1-11.2-UEFI    Boot from WEPE2.1.ISO----UEFI-FUWUQI-11.2
item WEPE2.1-11.2-BIOS    Boot from WEPE2.1.ISO----BIOS-FUWUQI-11.2

:WEPE2.1-UEFI
#echo WEPE2.1-UEFI
sanboot http://192.168.11.3/iso/WePE_64_V2.02019-0904.iso || goto failed
goto start

:WEPE2.1-BIOS
#echo WEPE2.1-BIOS
initrd http://192.168.11.3/iso/WePE_64_V2.02019-0904.iso
chain memdisk iso raw || goto failed
goto start

:WEPE2.1-11.2-UEFI
#echo WEPE2.1-UEFI
sanboot http://192.168.11.2/iso/WePE_64_V2.02019-0904.iso || goto failed
goto start

:WEPE2.1-11.2-BIOS
#echo WEPE2.1-BIOS
initrd http://192.168.11.2/iso/WePE_64_V2.02019-0904.iso
chain memdisk iso raw || goto failed
goto start
####################################

11.3 is openwrt 22.03.0-rc6
11.2 Windows server 2022 IIS I added MIMO type for iso file

The client I simulated with hyper-v
All turned off SafeBoot
First generation is BIOS, second generation is UEFI

In BIOS mode, both 11.2 and 11.3 can be started normally
However, in UEFI mode, 11.3 cannot be started and 11.2 can be started

What is the reason? Why does openwrt enable the BIOS to start but not uefi

**Comment by NiKiZe** [comment]
GitHub has formatting to make things more readable, please use code formatting for the script (you can edit your post)

This is unlikely an issue with iPXE, so moved to discussion so you can get help.

You did not say what happens or if any errors is shown. Just that it doesn't work. Without more information it is impossible to know.

In regards to booting windows Pe, please use wimboot instead. Memdisk works partially due to how the boot process is setup, but wimboot is much better.

You could have tried sanboot instead of memdisk in BIOS mode. But again don't expect that kind of setup to work.

**Reply by stappersg** [reply]
Yes, formatting helps expressing what is something about. Adding triple backticks, \`\`\`, before and after the relevant configuration will be an improvement.

This update is to express that I did read https://github.com/ipxe/ipxe/discussions/743#discussioncomment-3718887 as _"new community member reporting for duty"_ a message that said _"I have see an existing issue before I opened another issue"_. So now I say _"Welcome @chainofhonor"_     :smile: 

I look forward to read if [wimboot](https://ipxe.org/wimboot) solves the original problem.
